plugins/inputs/windows_event_log/wineventlog/utils.go

```diff
@@ -78,10 +79,21 @@ func CreateQuery(path string, levels []string) (*uint16, error) {
 }
 
 func UTF16ToUTF8Bytes(in []byte, length uint32) ([]byte, error) {
-	i := length
-
-	if length%2 != 0 {
-		i = length - 1
+	// Since Windows server 2022, the returned value of used buffer represents for double bytes char count,
+	// which is half of the actual buffer used by byte(what older Windows OS returns), checking if the length
+	//land on the end of used buffer, if no, double it.
+	var i int
+	if isTheEndOfContent(in, length) {
+		i = int(length)
+		if i%2 != 0 {
+			i--
+		}
+	} else {
+		log.Printf("W! Buffer used: %d is returning as double byte character count, doubling it for decoding", length)
```

Comment:
Should this really be warning?
If every single Windows Event on WinServer2022 will have this issue then it will be very noisy.

Reply:
I would agree. I'm not familiar with this code but I would be concerned about the frequency we emit this warn level log. I think that this should be removed entirely (after the tests/issue is confirmed to be fixed with these changes), or at the bare minimum moved to debug-level logging before merging this in.

Reply:
Yeah, thinking about that, to avoid the noise, maybe a debug is more suitable.